src/aws_request.erl

```diff
@@ -31,14 +22,17 @@ request(RequestFun, Options) ->
 
 %% Generate headers with an AWS signature version 4 for the specified
 %% request.
-sign_request(Client, Method, URL, Headers, Body) ->
+sign_request(Client, Method, URL, Headers0, Body) ->
     AccessKeyID = maps:get(access_key_id, Client),
     SecretAccessKey = maps:get(secret_access_key, Client),
     Region = maps:get(region, Client),
     Service = maps:get(service, Client),
     Token = maps:get(token, Client, undefined),
-    sign_request(AccessKeyID, SecretAccessKey, Region, Service, Token,
-                 Method, URL, Headers, Body).
+    Headers = case Token of
+                undefined -> Headers0;
+                _ -> [{<<"X-Amz-Security-Token">>, hackney_url:urlencode(Token)}|Headers0]
```

Comment:
Not a huge fan of `hackney_url:urlencode/1` here and to be 100% honest I'm not sure it's even needed but since my Client doesn't include a token for regular requests, I cannot test this. I've therefor kept it the same as in `aws_s3_presigned_url`. So see comment on that module.